packages/eas-cli/src/project/publish.ts

```diff
@@ -196,6 +210,43 @@ export function loadMetadata(distRoot: string): Metadata {
   return metadata;
 }
 
+/** Try to load the asset map for logging the names of assets published */
+export async function loadAssetMapAsync(distRoot: string): Promise<AssetMap | null> {
+  const assetMapPath = path.join(distRoot, 'assetmap.json');
+
+  if (!(await fs.pathExists(assetMapPath))) {
+    return null;
+  }
+
+  const assetMap: AssetMap = JsonFile.read(path.join(distRoot, 'assetmap.json'));
+  // TODO: basic validation?
+  return assetMap;
+}
+
+// exposed for testing
+export function getAssetHashFromPath(assetPath: string): string | null {
+  const [, hash] = assetPath.match(new RegExp(/assets\/([a-z0-9]+)$/, 'i')) ?? [];
+  return hash ?? null;
+}
+
+// exposed for testing
+export function getOriginalPathFromAssetMap(
+  assetMap: AssetMap | null,
+  asset: { path: string; ext: string }
+): string {
+  if (assetMap) {
+    const assetHash = getAssetHashFromPath(asset.path);
+    const assetMapEntry = assetHash && assetMap[assetHash];
+
+    if (assetMapEntry) {
+      const pathPrefix = assetMapEntry.httpServerLocation.substring('/assets'.length);
+      return `${pathPrefix}/${assetMapEntry.name}.${assetMapEntry.type}`;
+    }
+  }
+
+  return `${asset.path}.${asset.ext}`;
```

Comment:
should we return null in this case since we don't know the original path?

put another way, we should do the defaulting when we don't know at the time of display

Reply:
Sure, sounds good. Will change that.